Mathlib/Order/Antichain.lean

```diff
@@ -172,6 +173,28 @@ theorem isAntichain_singleton (a : α) (r : α → α → Prop) : IsAntichain r
 theorem Set.Subsingleton.isAntichain (hs : s.Subsingleton) (r : α → α → Prop) : IsAntichain r s :=
   hs.pairwise _
 
+/-- A set which is simultaneously a chain and antichain is subsingleton. -/
+lemma subsingleton_of_isChain_of_isAntichain (hs : IsChain r s) (ht : IsAntichain r s) :
+    s.Subsingleton := by
+  intro x hx y hy
+  by_contra! hne
+  cases hs hx hy hne with
+  | inl h => exact ht hx hy hne h
+  | inr h => exact ht hy hx hne.symm h
+
+lemma isChain_and_isAntichain_iff_subsingleton : IsChain r s ∧ IsAntichain r s ↔ s.Subsingleton :=
+  ⟨fun h ↦ subsingleton_of_isChain_of_isAntichain h.1 h.2, fun h ↦ ⟨h.isChain, h.isAntichain _⟩⟩
+
+/-- The intersection of a chain and an antichain is subsingleton.  -/
+lemma inter_subsingleton_of_isChain_of_isAntichain (hs : IsChain r s) (ht : IsAntichain r t) :
```

Comment:
This and the next theorem could use dot notation, e.g.
```suggestion
lemma IsChain.inter_subsingleton_of_isAntichain (hs : IsChain r s) (ht : IsAntichain r t) :
```

Reply:
It could, but I'm not convinced it's a good idea; there's no real reason for chain to be promoted over antichain here, which is why I went for the neutral version

Reply:
`Subsingleton.inter_of_isChain_of_isAntichain` would be another option.